app/src/main/java/com/amaze/filemanager/asynchronous/asynctasks/ReadFileTask.java

```diff
@@ -126,6 +127,9 @@ protected ReturnedValues doInBackground(Void... params) {
     } catch (IOException e) {
       e.printStackTrace();
       return new ReturnedValues(EXCEPTION_IO);
+    } catch (OutOfMemoryError e) {
```

Comment:
As I commented on telegram, catching OOM has issues, like the fact that you don't actually have any assurance of being able to create `new` objects.